### PR TITLE
[configure] Try stripping "--version-script" from LDSHAREDFLAGS if testing shared library support fails

### PR DESCRIPTION
**Pull request by mtl1979**
See #1748.

### CODE REVIEW

**Comment by haampie** [comment]
I didn't see this and submitted #1750, which is more reliable.

Better would be to test if `-Wl,--version-script` is supported at all.

File: configure [hunk 1]
```diff
@@ -600,10 +600,14 @@ int hello() {return getchar();}
 EOF
 if test $shared -eq 1; then
   printf "Checking for shared library support... " | tee -a configure.log
+  # Workaround for lld not liking --version-script
+  LDDSHAREDFLAGS="$(echo $LDSHAREDFLAGS | sed 's/,--version-script,[^ ]\+//')"
```
Comment (haampie) [inline configure:604]: I would not use sed to remove linker flags. It fails on: `-Wl,--version-script,...`, leaving `-Wl`, causing the compiler to complain about an unrecognized flag.

**Review by haampie — CHANGES_REQUESTED** [review]

File: configure (hunk 1 above)
Comment (mtl1979) [reply configure:604]: It's supposed to fail if `--version-script` is specified alone... In zlib-ng's configure script it is never specified alone.

**Comment by mtl1979** [comment]
> I didn't see this and submitted #1750, which is more reliable.
> 
> Better would be to test if `-Wl,--version-script` is supported at all.

True... But that would require writing "dummy" version script specific to the test, otherwise it would fail just like with the current version script.

**Pull request closed without merging**